File: modules/spring-core/testdata/java/inspection/conditionalOnPropertyValue/src/com/ConditionalOnPropertyConfiguration.java
package com;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.annotation.AliasFor;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.METHOD, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@ConditionalOnProperty
@interface ConditionalOnPropertyA {
    @AliasFor(annotation = ConditionalOnProperty.class, value = "value")
    String[] value() default {};
    
    @AliasFor(annotation = ConditionalOnProperty.class, value = "name")
    String[] nameEx() default {};
    
    @AliasFor(annotation = ConditionalOnProperty.class, value = "prefix")
    String prefixEx() default "";
}

@Configuration
public class DependsOnPropertyConfiguration {
    
    @ConditionalOnPropertyA("")
    @Bean
    String valid1() {return "";}
    
    @ConditionalOnPropertyA(nameEx = "")
    @Bean
    String valid2() {return "";}
    
    @ConditionalOnProperty
    @Bean
    String valueIsNotSet1() {return "";}
    
    @ConditionalOnPropertyA(prefixEx = "")
    @Bean
    String valueIsNotSet2() {return "";}
    
}

File: modules/spring-core/testdata/kotlin/inspection/conditionalOnPropertyValue/src/com/ConditionalOnPropertyConfiguration.kt
package com

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.annotation.AliasFor

@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.CLASS
)
@Retention(
    AnnotationRetention.RUNTIME
)
@ConditionalOnProperty
internal annotation class ConditionalOnPropertyA(
    @get:AliasFor(
        annotation = ConditionalOnProperty::class,
        value = "value"
    ) vararg val value: String = [],
    @get:AliasFor(
        annotation = ConditionalOnProperty::class,
        value = "name"
    ) val nameEx: Array<String> = [],
    @get:AliasFor(
        annotation = ConditionalOnProperty::class,
        value = "prefix"
    ) val prefixEx: String = ""
)

@Configuration
open class DependsOnPropertyConfiguration {
    @ConditionalOnPropertyA("")
    @Bean
    open fun valid1(): String {
        return ""
    }

    @ConditionalOnPropertyA(nameEx = [""])
    @Bean
    open fun valid2(): String {
        return ""
    }

    @ConditionalOnProperty
    @Bean
    open fun valueIsNotSet1(): String {
        return ""
    }

    @ConditionalOnPropertyA(prefixEx = "")
    @Bean
    open fun valueIsNotSet2(): String {
        return ""
    }
}